sparse/densification.py

```diff
@@ -0,0 +1,140 @@
+            raise ValueError('min_density must be a number between 0 and 1.')
+
+        self.densify = densify
+        self.max_size = int(max_size)
+        self.min_density = float(min_density)
+
+    def make_copy_of(self, other):
+        if isinstance(other, Iterable):
+            single = DensificationConfig.from_many(other)
+        else:
+            single = other
+
+        self.densify = single.densify
+        self.max_size = single.max_size
+        self.min_density = single.min_density
+
+    def _should_densify(self, size, density):
+        if self.densify:
+            return True
+        elif self.densify is False:
+            return False
+        else:
+            return self.max_size >= size or self.min_density <= density
+
+    def _raise_if_fails(self, size, density, name):
+        if not self._should_densify(size, density):
+            raise ValueError("Performing this operation would produce "
+                             "a dense result: %s" % name)
+
+    def check(self, name, *arrays):
+        result_shape = ()
+        density = 1.0
+
+        for arr in arrays:
+            if isinstance(arr, SparseArray):
+                density = min(density, arr.density)
+                result_shape = _get_broadcast_shape(result_shape, arr.shape)
+
+        size = np.prod(result_shape, dtype=np.uint64)
+
+        self._raise_if_fails(size, density, name)
+
+    @staticmethod
+    def validate(configs):
+        if isinstance(configs, Iterable):
+            for config in configs:
+                DensificationConfig._validate_single(config)
+
+            return
+
+        DensificationConfig._validate_single(configs)
+
+    @staticmethod
+    def _validate_single(config):
+        if not isinstance(config, DensificationConfig):
+            raise ValueError('Invalid DensificationConfig.')
+
+    @staticmethod
+    def combine(*configs):
+        result = set()
+
+        for config in configs:
+            if isinstance(config, Iterable):
+                DensificationConfig.validate(config)
+                result.update(config)
+            elif isinstance(config, DensificationConfig):
+                DensificationConfig._validate_single(config)
+                result.add(config)
+
+        return result
+
+    @staticmethod
+    def from_many(managers):
+        if isinstance(managers, DensificationConfig):
+            return DensificationConfig(managers)
+
+        densify = True
+        max_size = _max_size
+        min_density = 0.0
+
+        for manager in managers:
+            densify = _three_way_and(densify, manager.densify)
+            max_size = min(max_size, manager.max_size)
+            min_density = max(min_density, manager.min_density)
+
+        return DensificationConfig(densify, max_size, min_density)
+
+    def __str__(self):
+        if isinstance(self.densify, bool):
+            return '<DensificationConfig: densify=%s>' % self.densify
```

Comment:
I think there should be one unified `repr`.

Reply:
I feel otherwise... Printing fluff that isn't relevant to the repr can be useless. This just prints the required info.